src/main/java/bio/terra/service/filedata/google/gcs/GcsPdao.java

```diff
@@ -81,20 +83,43 @@ public GcsPdao(
     this.performanceLogger = performanceLogger;
   }
 
+  public Storage storageForBucket(GoogleBucketResource bucketResource) {
+    return gcsProjectFactory.getStorage(bucketResource.projectIdForBucket());
+  }
+
   /**
-   * Get all of the lines from any files matching the path, including wildcarded paths
+   * Get all of the lines from any files matching the path as a Stream, including wildcarded paths
    *
    * @param path path to files, or path including wildcard referring to many files
    * @param projectId Project ID to use for storage service in case of requester pays bucket
-   * @return All of the lines from all of the files matching the path
+   * @return All of the lines from all of the files matching the path, as a Stream
    */
-  public List<String> getGcsFilesLines(String path, String projectId) {
+  public Stream<String> getGcsFilesLinesStream(String path, String projectId) {
     Storage storage = gcsProjectFactory.getStorage(projectId);
     int lastWildcard = path.lastIndexOf("*");
     String prefixPath = lastWildcard >= 0 ? path.substring(0, lastWildcard) : path;
     return listGcsFiles(prefixPath, projectId, storage)
-        .flatMap(blob -> getGcsFileLines(blob, projectId, storage).stream())
-        .collect(Collectors.toList());
+        .flatMap(blob -> getBlobLinesStream(blob, projectId, storage));
+  }
+
+  @SuppressFBWarnings("OS_OPEN_STREAM")
+  private static Stream<String> getBlobLinesStream(Blob blob, String projectId, Storage storage) {
+    logger.info(String.format("Reading lines from %s", getGsPathFromBlob(blob)));
+    var reader = storage.reader(blob.getBlobId(), Storage.BlobSourceOption.userProject(projectId));
+    var channelReader = Channels.newReader(reader, StandardCharsets.UTF_8);
+    var bufferedReader = new BufferedReader(channelReader);
+    return bufferedReader.lines();
```

Comment:
@pshapiro4broad I've been toying with this all day and I can't seem to get it to not complain about an un-closed stream, specifically the `BufferedReader`. I'm unsure of how to pass the stream around while also making sure the `BufferedReader` gets closed.